src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift

```diff
@@ -434,6 +434,52 @@ final class HTTPClient: Client {
 		traceWarningPackageDownload(country: country, packageId: packageId, url: url, completion: completion)
 	}
 
+	func dccRegisterPublicKey(
+		isFake: Bool = false,
+		token: String,
+		publicKey: Data,
+		completion: @escaping DCCRegistrationCompletionHandler
+	) {
+
+		guard let request = try? URLRequest.dccPublicKeyRequest(
+			configuration: configuration,
+			token: token,
+			publicKey: publicKey,
+			headerValue: isFake ? 1 : 0
+		) else {
+			Log.error("Failed to create request")
+			completion(.failure(.urlCreationFailed))
+			return
+		}
+		session.response(for: request) { [weak self] result in
+			self?.queue.async {
+				switch result {
+				case let .success(response):
+					switch response.statusCode {
+					case 201:
+						completion(.success(()))
+					case 400:
+						completion(.failure(.badRequest))
+					case 403:
+						completion(.failure(.tokenNotAllowed))
+					case 404:
+						completion(.failure(.tokenDoesNotExist))
+					case 409:
+						completion(.failure(.tokenAlreadyAssigned))
+					case 500:
```

Comment:
Better expect more than just 500 – in case somebody learned how to assign proper response codes:
```suggestion
					case 500...599:
```

Reply:
500 is what the tec spec only wants